src/pages/Request.vue

```diff
@@ -27,11 +28,42 @@
 <script>
     import Config from '../config.js';
     import Store from '../store.js';
+
+    import mixin from '../mixins/mixins';
+    import constants from "../constants";
+
     export default {
         name: 'request',
+        mixins: [
+            mixin.messages
+        ],
         data: function () {
             return {
-                username: null
+                username: null,
+                netMgr: Store.netMgr
+            }
+        },
+        methods: {
+            isEmailValid: function (email) {
+                var re = /\S+@\S+/;
+                return re.test(email);
+            },
+            onRequestResetEmail: function () {
+                if (this.isEmailValid(this.username)) {
+
+                    return this.netMgr.apiPost('user/lost_password', {'email' : this.username},
+
+                        function (response) {
+                            this.setMessage(response.data[1], constants.MESSAGE_SUCCESS)
```

Comment:
Would it be better if we keyed this data to make more clear what it is? Happy for long grass and a comment.